nailgun/entity_mixins.py

```diff
@@ -551,17 +552,30 @@ def to_json_dict(self):
         json_dct = {}
         for field_name, field in fields.items():
             if field_name in values:
-                if isinstance(field, OneToOneField):
-                    json_dct[field_name] = values[field_name].to_json_dict()
+                value = values[field_name]
+                if value is None:
+                    json_dct[field_name] = None
+                elif isinstance(field, OneToOneField):
+                    json_dct[field_name] = value.to_json_dict()
                 elif isinstance(field, OneToManyField):
                     json_dct[field_name] = [
-                        entity.to_json_dict() for entity in values[field_name]
+                        entity.to_json_dict() for entity in value
                     ]
                 else:
-                    json_dct[field_name] = to_json_serializable(
-                        values[field_name])
+                    json_dct[field_name] = to_json_serializable(value)
```

Comment:
Finnaly found the case I mentioned above:

```python
hostgroup = entities.HostGroup(
            location=[self.loc],
            organization=[self.org],
        ).create()
        hostgroup.to_json_dict()
```
Stack trace:

```console
    def to_json_dict(self):
        """Create a dct with Entity properties for json encoding.
            It can be overridden by subclasses for each standard serialization
            doesn't work. By default it call _to_json_dict on OneToOne fields
            and build a list calling the same method on each object on OneToMany
            fields.
    
            :return: dct
            """
        fields, values = self.get_fields(), self.get_values()
        json_dct = {}
        for field_name, field in fields.items():
            if field_name in values:
                if isinstance(field, OneToOneField):
>                   json_dct[field_name] = values[field_name].to_json_dict()
E                   AttributeError: 'NoneType' object has no attribute 'to_json_dict'
```